be/src/exec/vectorized/aggregate/agg_hash_map.h

```diff
@@ -656,26 +653,26 @@ struct AggHashMapWithSerializedKeyFixedSize {
 
         not_founds->assign(chunk_size, 0);
 
-        FixedSizeSliceKey key;
-
-        if (!has_null_column) {
+        if (has_null_column) {
             for (size_t i = 0; i < chunk_size; ++i) {
-                memcpy(key.u.data, buffer + i * max_fixed_size, max_fixed_size);
-                if (auto iter = hash_map.find(key); iter != hash_map.end()) {
-                    (*agg_states)[i] = iter->second;
-                } else {
-                    (*not_founds)[i] = 1;
-                }
+                caches[i].key.u.size = slice_sizes[i];
             }
-        } else {
-            for (size_t i = 0; i < chunk_size; ++i) {
-                memcpy(key.u.data, buffer + i * max_fixed_size, max_fixed_size);
-                key.u.size = slice_sizes[i];
-                if (auto iter = hash_map.find(key); iter != hash_map.end()) {
-                    (*agg_states)[i] = iter->second;
-                } else {
-                    (*not_founds)[i] = 1;
-                }
+        }
+        for (size_t i = 0; i < chunk_size; i++) {
+            caches[i].hashval = hash_map.hash_function()(caches[i].key);
+        }
+
+        size_t __prefetch_index = AGG_HASH_MAP_DEFAULT_PREFETCH_DIST;
+
+        for (size_t i = 0; i < chunk_size; ++i) {
+            if (__prefetch_index < chunk_size) {
+                hash_map.prefetch_hash(caches[__prefetch_index++].hashval);
+            }
+            FixedSizeSliceKey& key = caches[i].key;
+            if (auto iter = hash_map.find(key); iter != hash_map.end()) {
```

Comment:
there is a method called `find(key, hashval)`, maybe we can use that?

Reply:
good catch. thanks.